build.rs

```diff
@@ -1,8 +1,54 @@
-extern crate cc;
+use color_eyre::{
+    eyre::{eyre, Context, Report, Result},
+    Help, SectionExt,
+};
+use std::{env, fs, path::PathBuf, process::Command};
 
-use std::env;
+trait CommandExt {
+    /// wrapper for `status` fn on `Command` that constructs informative error
+    /// reports
+    fn status2(&mut self) -> Result<(), Report>;
+
+    /// wrapper for `output` fn on `Command` that constructs informative error
+    /// reports
+    fn output2(&mut self) -> Result<String, Report>;
+}
+
+#[cfg(feature = "generate")]
+fn bindgen_headers() -> Result<()> {
+    let bindings = bindgen::Builder::default()
+        .header("depend/zcash/src/script/zcashconsensus.h")
+        // Tell cargo to invalidate the built crate whenever any of the
+        // included header files changed.
+        .parse_callbacks(Box::new(bindgen::CargoCallbacks))
+        // Finish the builder and generate the bindings.
+        .generate()
+        .ok_or("Unable to generate bindings")?;
+
+    // Write the bindings to the $OUT_DIR/bindings.rs file.
+    let out_path = env::var("OUT_DIR")?;
+    let out_path = PathBuf::from(out_path);
+    bindings
+        .write_to_file(out_path.join("bindings.rs"))
+        .wrap_err("unable to write bindings")?;
+
+    Ok(())
+}
+
+#[cfg(not(feature = "generate"))]
```

Comment:
I don't think it's safe to ship pre-generated bindings since bindings are generated for a specific ABI. `bindgen` is "safe" because it uses `libclang` (which we trust to be ABI-compatible with the C/C++ compiler) to query the size, alignment, etc. of types and various other ABI-dependent properties.

Hand-reviewing these bindings, they use `std::os::raw` in most places and would be largely portable for that reason. However, they include `pub type zcashconsensus_error_t = u32;` which isn't portable (since the type of an `enum` is implementation-defined).

Reply:
That sucks, I really didn't want to require CI to generate the bindings because bindgen is missing clang deps in CI :/ I'll go ahead and figure out the right command to install those